File: cpp/342_4的幂_easy.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n <= 0 || n & (n-1)) return false;
        int cnt = 0;
        while (n) {
            cnt++;
            n = n >> 1;
        }
        return cnt % 2 ? true : false;
    }
};

File: python/342_4的幂_easy.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0 or n & (n - 1):
            return False
        cnt = 0
        while n:
            cnt += 1
            n = n >> 1
        return True if cnt % 2 == 1 else False